src/main/kotlin/no/nav/familie/ks/sak/sikkerhet/TilgangService.kt

```diff
@@ -0,0 +1,183 @@
+import no.nav.familie.ks.sak.kjerne.personopplysninggrunnlag.domene.PersonopplysningGrunnlagRepository
+import org.springframework.cache.CacheManager
+import org.springframework.stereotype.Service
+
+@Service
+class TilgangService(
+    private val fagsakService: FagsakService,
+    private val behandlingRepository: BehandlingRepository,
+    private val personopplysningGrunnlagRepository: PersonopplysningGrunnlagRepository,
+    private val rolleConfig: RolleConfig,
+    private val integrasjonService: IntegrasjonService,
+    private val cacheManager: CacheManager,
+    private val auditLogger: AuditLogger
+) {
+
+    /**
+     * Sjekk om saksbehandler har tilgang til å gjøre en gitt handling.
+     *
+     * @param minimumBehandlerRolle den laveste rolle som kreves for den angitte handlingen
+     * @param handling kort beskrivelse for handlingen. Eksempel: 'endre vilkår', 'oppprette behandling'.
+     * Handlingen kommer til saksbehandler så det er viktig at denne gir mening.
+     */
+    fun validerTilgangTilHandling(minimumBehandlerRolle: BehandlerRolle, handling: String) {
+        val høyesteRolletilgang = SikkerhetContext.hentHøyesteRolletilgangForInnloggetBruker(rolleConfig)
+        if (minimumBehandlerRolle.nivå > høyesteRolletilgang.nivå) {
+            throw RolleTilgangskontrollFeil(
+                melding = "${SikkerhetContext.hentSaksbehandlerNavn()} med rolle $høyesteRolletilgang " +
+                    "har ikke tilgang til å $handling. Krever $minimumBehandlerRolle."
+            )
+        }
+    }
+
+    /**
+     * Sjekk om saksbehandler har tilgang til å behandle bestemte personidenter
+     * @param personIdenter liste over person identer man skal sjekke tilgang til
+     * @param event operasjon som skal gjøres med identene
+     * @param minimumBehandlerRolle den laveste rolle som kreves for den angitte handlingen
+     * @param handling kort beskrivelse for handlingen.
+     */
+    fun validerTilgangTilPersoner(
+        personIdenter: List<String>,
+        event: AuditLoggerEvent,
+        minimumBehandlerRolle: BehandlerRolle,
+        handling: String
+    ) {
+        validerTilgangTilHandling(minimumBehandlerRolle, handling)
+        loggPersonoppslag(personIdenter, event)
+        if (!harTilgangTilPersoner(personIdenter)) {
+            throw RolleTilgangskontrollFeil(
+                melding = "Saksbehandler ${SikkerhetContext.hentSaksbehandler()} " +
+                    "har ikke tilgang til å behandle $personIdenter"
+            )
+        }
+    }
+
+    /**
+     * Sjekk om saksbehandler har tilgang til å gjøre bestemt handling og om saksbehandler kan behandle behandling
+     * @param behandlingId id til behandling det skal sjekkes tilgang til
+     * @param event operasjon som skal gjøres med identene
+     * @param minimumBehandlerRolle den laveste rolle som kreves for den angitte handlingen
+     * @param handling kort beskrivelse for handlingen.
+     */
+    fun validerTilgangTilBehandling(
```

Comment:
Synes det er rart at vi validerer tilgang på behandlingsnivå. Man må jo via fagsak for å komme til behandling, så den eneste måten denne valideringen er relevant er om man sniper urlen til sak som allerede er opprettet som ikke har tatt med seg fortrolige personer. 

Lurer også litt på hvordan det påvirker feks. registrere søknad-steget der man kan dra inn barn som ikke er i søknaden 🤔